server/src/repositories/test_repository.ts

```diff
@@ -27,4 +28,46 @@ export class TestRepository {
             ])
             .finally(() => client.release());
     }
+
+    async findTestByTestId(testId: number): Promise<TestResultWithAddress> {
+        const client = await this.pool.connect();
+        const queryString = `SELECT *
+                             FROM test_results AS tr
+                             WHERE tr.test_id = $1`;
+
+        const res = await client.query(queryString, [testId]).finally(async () => client.release());
+        const row = res.rows[0];
+        if (!row) {
+            return null;
+        }
+        const address = await this.findAddressById(row.address_id);
+        return {
+            patientId: row.patient_id,
+            result: row.result,
+            testDate: row.test_date,
+            testType: row.test_type,
+            ...address,
+        };
+    }
+
+    async findAddressById(addressId: number): Promise<Address> {
+        const client = await this.pool.connect();
+        const queryString = `SELECT *
+                             FROM addresses AS a
+                             WHERE a.address_id = $1`;
+        const res = await client.query(queryString, [addressId]).finally(async () => client.release());
+        const row = res.rows[0];
+        if (!row) {
+            return null;
+        }
+        return {
+            addressId: row.address_id,
+            streetAddress: row.street_address,
+            streetAddressLineTwo: row.street_address_line_two,
+            city: row.city,
+            province: row.province,
+            postalCode: row.postal_code,
+            country: row.country,
+        };
+    }
```

Comment:
If you want to leave it as its own function then this should be in the user repository / refactor all the address functions to a new address repository. You can also just do a join in the query where you get the test result to also get the address row

Reply:
yea a join makes sense honestly don't really know what i was thinking